### PR TITLE
More tests and some cleanups of BibtexKeyPatternUtil

### PR DESCRIPTION
**Pull request by oscargus**
Removed some unused/unnecessary code and added a number of tests.

- [x] Tests created for changes

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtil.java [hunk 1]
```diff
@@ -692,7 +665,7 @@ else if (val.matches("edtr\\d+")) {
                 String ss = entry.getFieldOrAlias(FieldName.YEAR).orElse("");
                 if (ss.isEmpty()) {
                     return ss;
-                } else if (ss.startsWith("in") || ss.startsWith("sub")) {
+                } else if (ss.startsWith("in") || ss.startsWith("sub")) { // TODO: Year starting with in or sub? Why?
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtil.java]: No idea why sub or in either. Any idea why the var is called "ss"? I would change it to a more expressive variable name

**Comment by Siedlerchr** [comment]
If you change the one variable name to a more expresive one, I'm fine :+1:

File: src/test/java/net/sf/jabref/logic/bibtexkeypattern/MakeLabelWithDatabaseTest.java [hunk 2]
```diff
@@ -285,4 +285,131 @@ public void generateKeyTitleAbbr() {
...
+        metadata.setBibtexKeyPattern(bibtexKeyPattern);
+        BibEntry entry = new BibEntry();
+        entry.setField("title", "An awesome paper on JabRef");
+        BibtexKeyPatternUtil.makeLabel(metadata, database, entry, preferences);
+        assertEquals(Optional.of("awesomepaperJabRef"), entry.getCiteKeyOptional());
+    }
+
+    @Test
+    public void generateKeyVeryshorttitle() {
+        preferences = new BibtexKeyPatternPreferences("", "", false, true, false, pattern);
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/bibtexkeypattern/MakeLabelWithDatabaseTest.java]: Move som of this initialization code to the setup method.

**Comment by tobiasdiez** [comment]
I second @Siedlerchr's opinion: LGTM. Just move some of the test code to `@Setup` and change the variable name, then this can be merged. (I thus remove the ready-for-review tag)

File: src/main/java/net/sf/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtil.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtil.java]: Ahh, I think I understand it now: `IP` = in press/preparation, `in` = in press/preparation, and `sub` = submitted. Better to use the note field for that in general I guess, but maybe worthwhile keeping for compatibility. (One may even add support `under review` leading to either `IP` or `UR`.)

**Pull request merged**